modules/web/js/ballerina/model/tree/compilation-unit-node.js

```diff
@@ -20,10 +20,32 @@ import _ from 'lodash';
 import log from 'log';
 import AbstractCompilationUnitNode from './abstract-tree/compilation-unit-node';
 import TreeUtils from '../tree-util';
-
+import TreeBuilder from '../tree-builder';
+import { parseContent } from '../../../api-client/api-client';
 
 class CompilationUnitNode extends AbstractCompilationUnitNode {
 
+    constructor() {
+        super();
+        this.on('tree-modified', (e) => {
+            if (e.type !== 'child-added') {
+                return;
+            }
+
+            if (TreeUtils.isAssignment(e.data.node) && TreeUtils.isInvocation(e.data.node.getExpression())) {
+                const fullPackageName = e.data.node.getExpression().getFullPackageName();
+
+                const value = 'import ' + fullPackageName + ';\n';
+                parseContent(value)
+                    .then((jsonTree) => {
+                        if (jsonTree.model.topLevelNodes[0]) {
+                            this.addImport(TreeBuilder.build(jsonTree.model.topLevelNodes[0]));
```

Comment:
I don't think its a good idea for the Tree to build itself, this leads to circular dependency like we had earlier. why cant we keep the builder separate.